resteasy-core/src/main/java/org/jboss/resteasy/core/MediaTypeMap.java

```diff
@@ -576,14 +577,14 @@ private void mergeEverything(Entry<T> entry) {
       cached = convert(matches);
       if (useCache) {
          Map<CachedMediaTypeAndClass, List<T>> cache = classCache;
-         if (classCache == null) {
+         if (cache == null) {
             synchronized (this)
             {
                if (classCache == null)

```

Comment:
if (cache== null) check again here ?

Reply:
I think that it's not needed to repeat the check for the cache var, cache is local to the method, so if it was null then it's null now. Only classCache can be initialized by other threads.